### PR TITLE
Expose Gibbs free energy

### LINKED ISSUE #397: Gibbs Free Energy
Given we have an entropy call, it seems relatively trivial now to add a Gibbs free energy call to the EOS base class that automatically converts between potentials.

$$G(\rho, T) = \varepsilon(\rho, T) + \frac{P(\rho, T)}{\rho} - T S(\rho, T)$$

### PR DESCRIPTION
**Pull request by Yurlungur**
## PR Summary

This PR exposes the Gibbs free energy. I do this by transforming from the entropy in the base class. But of course, we could instead do the opposite. Individual EOS's may override the defaults but I only implemented the defaults in the base class and variant. Down the line we will need to expose proper passthroughs for EOSPAC and the tabulated EOS's. I also add missing documentation for entropy and free energy.  This MR resolves #397 .

## PR Checklist

- [x] Adds a test for any bugs fixed. Adds tests for new features.
- [x] Format your changes by using the `make format` command after configuring with `cmake`.
- [x] Document any new features, update documentation for changes made.
- [x] Make sure the copyright notice on any files you modified is up to date.
- [x] After creating a pull request, note it in the CHANGELOG.md file.
- [ ] LANL employees: make sure tests pass both on the github CI and on the Darwin CI

If preparing for a new release, in addition please check the following:
- [ ] Update the version in cmake.
- [ ] Move the changes in the CHANGELOG.md file under a new header for the new release, and reset the categories.
- [ ] Ensure that any `when='@main'` dependencies are updated to the release version in the package.py

### CODE REVIEW

File: singularity-eos/eos/eos_base.hpp [hunk 1]
```diff
@@ -184,6 +186,28 @@ class EosBase {
...
+    Real P = copy.PressureFromDensityTemperature(rho, T, lambda);
+    Real S = copy.EntropyFromDensityTemperature(rho, T, lambda);
+    return sie + (P / rho) - T * S;
+  }
+  template <typename Indexer_t = Real *>
+  PORTABLE_INLINE_FUNCTION Real GibbsFreeEnergyFromDensityInternalEnergy(
+      const Real rho, const Real sie,
+      Indexer_t &&lambda = static_cast<Real *>(nullptr)) const {
+    const CRTP copy = *(static_cast<CRTP const *>(this));
+    Real T = copy.TemperatureFromDensityInternalEnergy(rho, T, lambda);
```
Comment (aematts) [inline singularity-eos/eos/eos_base.hpp:205]: I think this should be sie, not T in the argument list.

File: test/test_eos_ideal.cpp [hunk 2]
```diff
@@ -54,6 +54,14 @@ SCENARIO("Ideal gas entropy", "[IdealGas][Entropy]") {
...
         INFO("Entropy: " << entropy << "  True entropy: " << entropy_true);
         CHECK(isClose(entropy, entropy_true, 1e-12));
+
+        AND_THEN("The free energy agrees") {
+          const Real sie = host_eos.InternalEnergyFromDensityTemperature(rho, T);
+          const Real P = host_eos.PressureFromDensityTemperature(rho, T);
+          const Real G_true = sie + (P / rho) - T * entropy;
+          const Real G = host_eos.GibbsFreeEnergyFromDensityTemperature(rho, T);
+          CHECK(isClose(G, G_true, 1e-12));
+        }
```
Comment (aematts) [inline test/test_eos_ideal.cpp:64]: Maybe the same test with DensityInternalEnergy? Then you would have seen the error above ;).

File: test/test_eos_ideal.cpp [hunk 3]
```diff
@@ -32,7 +32,7 @@
 using singularity::IdealGas;
 using EOS = singularity::Variant<IdealGas>;
 
-SCENARIO("Ideal gas entropy", "[IdealGas][Entropy]") {
+SCENARIO("Ideal gas entropy", "[IdealGas][Entropy][FreeEnergy]") {
```
Comment (aematts) [inline test/test_eos_ideal.cpp:35]: FreeEnergy can be Gibbs or Helmholtz. Maybe GFreeEnergy?

**Review by aematts — CHANGES_REQUESTED** [review]

File: singularity-eos/eos/eos_base.hpp (hunk 1 above)
Comment (Yurlungur) [reply singularity-eos/eos/eos_base.hpp:205]: Oops! Thanks for the catch!

File: test/test_eos_ideal.cpp (hunk 2 above)
Comment (Yurlungur) [reply test/test_eos_ideal.cpp:64]: Yep good idea. I added both now.

File: test/test_eos_ideal.cpp (hunk 3 above)
Comment (Yurlungur) [reply test/test_eos_ideal.cpp:35]: Changed to `GibbsFreeEnergy` I don't see any reason not to be verbose here.

**Comment by Yurlungur** [comment]
Thanks for the useful feedback, @aematts ! I addressed your comments. Can you re-review and approve at your convenience?

**Review by aematts — APPROVED** [review]
Go!

**Comment by jhp-lanl** [comment]
Thanks @Yurlungur ! This looks good to me. Thanks for adding the note about EOSPAC

**Pull request merged**